sos/collector/sosnode.py

```diff
@@ -0,0 +1,882 @@
+            if ret:
+                self.ui_msg('Successfully collected sosreport')
+                self.file_list.append(self.sos_path.split('/')[-1])
+            else:
+                self.log_error('Failed to retrieve sosreport')
+                raise SystemExit
+            self.hash_retrieved = self.retrieve_file(self.sos_path + '.md5')
+            if self.hash_retrieved:
+                self.file_list.append(self.sos_path.split('/')[-1] + '.md5')
+            return True
+        else:
+            # sos sometimes fails but still returns a 0 exit code
+            if self.stderr.read():
+                e = self.stderr.read()
+            else:
+                e = [x.strip() for x in self.stdout.readlines() if x.strip][-1]
+            self.soslog.error(
+                'Failed to run sosreport on %s: %s' % (self.address, e))
+            self.log_error('Failed to run sosreport. %s' % e)
+            return False
+
+    def remove_sos_archive(self):
+        """Remove the sosreport archive from the node, since we have
+        collected it and it would be wasted space otherwise"""
+        if self.sos_path is None:
+            return
+        if 'sosreport' not in self.sos_path:
+            self.log_debug("Node sosreport path %s looks incorrect. Not "
+                           "attempting to remove path" % self.sos_path)
+            return
+        removed = self.remove_file(self.sos_path)
+        if not removed:
+            self.log_error('Failed to remove sosreport')
+
+    def cleanup(self):
+        """Remove the sos archive from the node once we have it locally"""
+        self.remove_sos_archive()
+        if self.hash_retrieved:
+            self.remove_file(self.sos_path + '.md5')
+        cleanup = self.host.set_cleanup_cmd()
+        if cleanup:
+            self.run_command(cleanup)
+
+    def collect_extra_cmd(self, filenames):
+        """Collect the file created by a cluster outside of sos"""
+        for filename in filenames:
+            try:
+                if self.need_sudo or self.opts.become_root:
+                    try:
+                        self.make_archive_readable(filename)
+                    except Exception as err:
+                        self.log_error("Unable to retrieve file %s" % filename)
+                        self.log_debug("Failed to make file %s readable: %s"
+                                       % (filename, err))
+                        continue
+                ret = self.retrieve_file(filename)
+                if ret:
+                    self.file_list.append(filename.split('/')[-1])
+                    self.remove_file(filename)
+                else:
+                    self.log_error("Unable to retrieve file %s" % filename)
+            except Exception as e:
+                msg = 'Error collecting additional data from master: %s' % e
+                self.log_error(msg)
+
+    def make_archive_readable(self, filepath):
```

Comment:
`os.chmod(filepath, stat.S_IROTH)` doesnt work?

Reply:
Unfortunately it does not on remote nodes.